add: calendar

Ryzhon opened the pull request:
カレンダー機能を実装しました。

Commit:
add: calendar

Commit:
rubocop

Ryzhon commented:
先にmainで編集してからプッシュしてしまったため、カレンダーの差分を消すことができませんでした。
申し訳ございません。

Commit:
Merge branch 'main' into calendar

JunichiIto commented on 02.calendar/cal.rb:
```diff
@@ -0,0 +1,25 @@
+
+params = ARGV.getopts(' ', "y:#{Date.today.year}", "m:#{Date.today.month}")
+
+current_date = Date.today
```
```suggestion
today = Date.today
```

の方がシンプルでいいかもしれません

Ryzhon replied:
おっしゃる通りだと思います。
todayに変更します。

JunichiIto commented on 02.calendar/cal.rb:
```diff
@@ -0,0 +1,25 @@
+params = ARGV.getopts(' ', "y:#{Date.today.year}", "m:#{Date.today.month}")
+
+current_date = Date.today
+year = params[:y] || current_date.year
```
5行目で `Date.today.year` を指定しているので `|| current_date.year` と内容が重複しているような？ 🤔 
どっちかが不要だったりしませんか？

あとyearとmonthの寿命が長いので短くしましょう。
https://bootcamp.fjord.jp/pages/311

Ryzhon replied:
改めて見てみるとgetoptsでのデフォルト値の設定は見にくかったので、paramが入っていない場合に今日の日付を入れるように修正しました。
また、year, monthは一番上で定義する必要はないので、実行直前に修正しました。

JunichiIto commented on 02.calendar/cal.rb:
```diff
@@ -0,0 +1,25 @@
+  print "\n"
+end
+
+print_calendar(year, month)
```
ファイル末尾の赤丸が出ないようにエディタを設定してください〜。
https://docs.komagata.org/5676#crlf

Ryzhon replied:
vscodeの`insertFinalNewline`をtrueに変更しました。

JunichiIto reviewed (commented):
コメントしました〜。

Commit:
fix: デフォルト値とライフスパンを短くする
改めて見てみるとgetoptsでのデフォルト値の設定は見にくかったので、paramが入っていない場合に今日の日付を入れるように修正しました。
また、year, monthは一番上で定義する必要はないので、実行直前に修正しました。

JunichiIto commented on 02.calendar/cal.rb:
```diff
@@ -0,0 +1,28 @@
+  print "\n"
+end
+
+year = params['y'] ? params['y'].to_i : today.year
```
```suggestion
params = ARGV.getopts('', 'y:', 'm:')
today = Date.today
year = params['y'] ? params['y'].to_i : today.year
```

変数の寿命を短くするならparamsとtodayもここにないとダメですね

あと高等テクニックですがこんな書き方もできます。

```ruby
year = params['y']&.to_i || today.year
```

Ryzhon replied:
なるほど。そんな書き方もあるんですね。
勉強になります！修正してからマージします！

JunichiIto reviewed (approved):
1点コメントしましたが、大きな問題ではないのでこれでOKです！

Commit:
fix: 可読性向上
